kv.go

```diff
@@ -10,83 +10,159 @@ import "context"
 // different behavior.
 var DefaultIterOptions = IterOptions{}
 
+// IterOptions contains the full set of available iterator options,
+// it can be provided when creating an [Iterator] from a [Store].
 type IterOptions struct {
-	// Prefix iteration, all keys *must* contain
-	// the designated prefix. Prefix takes precendent
-	// over range (start/end). If Prefix is nil, range
-	// will be used (regardless if start/end is nil)
+	// Prefix iteration, only keys beginning with the designated prefix
+	// with the given prefix will be yielded.
+	//
+	// Keys exactly matching the provided `Prefix` value will not be
+	// yielded.
+	//
+	// Providing a Prefix value should cause the Start and End options
+	// to be ignored, although this is currently untested:
+	// https://github.com/sourcenetwork/corekv/issues/35
 	Prefix []byte
 
-	// Range iteration from start (inclusive) to end explusive.
-	// Start must be lexigraphically less
-	// then end, unless `nil` is used.
+	// If Prefix is nil, and Start is provided, the iterator will
+	// only yield items with a key lexographically greater than or
+	// equal to this value.
 	//
-	// Start defined as nil will iterate from the first
-	// key in the store.
-	// End defined as nil will iterate till the last key
-	// in the store.
+	// Providing an `End` value equal to or smaller than this value
+	// will result in undefined behaviour:
+	// https://github.com/sourcenetwork/corekv/issues/32
 	Start []byte
-	End   []byte
 
-	// Reverse the direction of the iteration.
+	// If Prefix is nil, and Start is provided, the iterator will
+	// only yield items with a key lexographically smaller than this
+	// value.
+	//
+	// Providing an End value equal to or smaller than Start
+	// will result in undefined behaviour:
+	// https://github.com/sourcenetwork/corekv/issues/32
+	End []byte
```

Comment:
todo: This is the same documentation as `Start`. It should be specific to `End`

Reply:
Nearly the same, nice spot lol - I flipped `greater than or equal` to `smaller` but forgot to swap `Start` for `End` in that first sentence :grin:

I think last sentence can be the same for both.

- [ ] Fix copy-paste problem